packages/extension/src/companion/CompanionContent.tsx

```diff
@@ -19,6 +19,7 @@ import AnalyticsContext from '@dailydotdev/shared/src/contexts/AnalyticsContext'
 import { CompanionEngagements } from './CompanionEngagements';
 import { CompanionDiscussion } from './CompanionDiscussion';
 import { useBackgroundPaginatedRequest } from './useBackgroundPaginatedRequest';
+import { getCompanionWrapper } from './common';
```

Comment:
Have you considered this context we chatted about?
What where the implications, would love to hear about this.

Reply:
Yes, but we raised it here: https://dailydotdev.slack.com/archives/C02E2C3C13R/p1679929905819559?thread_ts=1679929715.768539&cid=C02E2C3C13R

Reply:
After debugging the changes, there is not much difference between the usages. Both use cases must still live under a React component.
Current:
```
  const { requestMethod } = useRequestProtocol();
```
With Context:
```
  const { requestMethod } = useContext(RequestContext);
```

With the changes above there seem to be no impact on performance too and when an where we can use it.

Reply:
Let's maybe take some time today to chat about the implications and what we are trying to solve with this maybe?